### PR TITLE
Allow different timeouts per configuration instance, using HttpUtils as default

### PR DESCRIPTION
**Pull request by papegaaij**
This PR restores the read and connect timeout settings in OAuthConfiguration and OidcConfiguration. HttpUtils now serves as default.

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/util/HttpUtils.java [hunk 1]
```diff
@@ -18,9 +18,9 @@
  */
 public final class HttpUtils {
 
-    private static int connectTimeout = HttpConstants.DEFAULT_CONNECT_TIMEOUT;
+    private static int defaultConnectTimeout = HttpConstants.DEFAULT_CONNECT_TIMEOUT;
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/util/HttpUtils.java]: I'm not sure to understand why we now have default timeouts instead of timeouts...

**Comment by leleuj** [comment]
The `HttpUtils` class should not be changed. Only timeouts should be added back for the OAuth and OpenID Connect classes.

**Comment by papegaaij** [comment]
I decided to change it this way to prevent a change in behavior when people already changed their code to use HttpUtils.

**Comment by papegaaij** [comment]
I can change the PR to leave HttpUtils untouched and initialize the timeouts to the constants, but that would break existing applications using HttpUtils to set the timeout.

**Comment by leleuj** [comment]
In fact, this change was already a breaking one (as all timeouts were set to one value), which is something we do not allow for minor versions.
So we definitely need to revert this change as it was in 2.0.

**Comment by papegaaij** [comment]
I've rewritten the original commit (so the change still is only 1 commit) to only change the configuration classes.

**Comment by leleuj** [comment]
Thanks. We'll see if other people report an issue with a unique timeout when testing the version 2.1.

**Pull request merged**